src/coreComponents/physicsSolvers/multiphysics/ThermalSinglePhasePoromechanicsKernel.hpp

```diff
@@ -0,0 +1,606 @@
+                    real64 const & dPorosity_dTemperature,
+                    real64 const & dSolidDensity_dPressure,
+                    StackVariables & stack ) const
+  {
+    Base::computeBodyForce( k, q,
+                            porosity,
+                            dPorosity_dVolStrain,
+                            dPorosity_dPressure,
+                            dPorosity_dTemperature,
+                            dSolidDensity_dPressure,
+                            stack );
+
+    real64 const dMixtureDens_dTemperature =
+      dPorosity_dTemperature * ( -m_solidDensity( k, q ) + m_fluidDensity( k, q ) )
+      + porosity * m_dFluidDensity_dTemperature( k, q );
+
+    LvArray::tensorOps::scaledCopy< 3 >( stack.dBodyForce_dTemperature, m_gravityVector, dMixtureDens_dTemperature );
+  }
+
+  /**
+   * @brief Helper function to compute the fluid mass/energy increment and its derivatives wrt primary variables
+   * @param[in] k the element index
+   * @param[in] q the quadrature point index
+   * @param[in] porosity the element porosity
+   * @param[in] porosity_n the element porosity at the previous converged time step
+   * @param[in] dPorosity_dVolStrain the derivative of porosity wrt volumetric strain increment
+   * @param[in] dPorosity_dPressure the derivative of porosity wrt pressure
+   * @param[in] dPorosity_dTemperature the derivative of porosity wrt temperature
+   * @param[inout] stack the stack variables
+   */
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  void
+  computeFluidIncrement( localIndex const k,
+                         localIndex const q,
+                         real64 const & porosity,
+                         real64 const & porosity_n,
+                         real64 const & dPorosity_dVolStrain,
+                         real64 const & dPorosity_dPressure,
+                         real64 const & dPorosity_dTemperature,
+                         StackVariables & stack ) const
+  {
+    // Step 1: compute fluid mass increment and its derivatives wrt vol strain and pressure
+    Base::computeFluidIncrement( k, q,
+                                 porosity,
+                                 porosity_n,
+                                 dPorosity_dVolStrain,
+                                 dPorosity_dPressure,
+                                 dPorosity_dTemperature,
+                                 stack );
+
+    // Step 2: compute derivative of fluid mass increment wrt temperature
+    stack.dFluidMassIncrement_dTemperature = dPorosity_dTemperature * m_fluidDensity( k, q ) + porosity * m_dFluidDensity_dTemperature( k, q );
+
+    // Step 3: compute fluid energy increment and its derivatives wrt vol strain, pressure, and temperature
+    real64 const fluidMass = porosity * m_fluidDensity( k, q );
+    real64 const fluidEnergy = fluidMass * m_fluidInternalEnergy( k, q );
+    real64 const fluidEnergy_n = porosity_n * m_fluidDensity_n( k, q ) * m_fluidInternalEnergy_n( k, q );
```

Comment:
Here it is not really necessary to add a `fluidMass_n` because it is computed once. I added `fluidMass` because it is used multiple times in this function.